[BUG] ESM/CJS error with `@refinedev/devtools`

### Describe the bug

Hey there again.

Seems like Ive found another  ESM/CJS issue
```sh
⨯ file:///Users/Projects/Code/app/node_modules/@refinedev/devtools/dist/index.mjs:2

SyntaxError: Named export 'getParentOfFiber' not found. The requested module '@aliemir/dom-to-fiber-utils' is a CommonJS module, which may not support all module.exports as named exports.
CommonJS modules can always be imported via the default export, for example using:

import pkg from '@aliemir/dom-to-fiber-utils';
  at ModuleJob._instantiate (node:internal/modules/esm/module_job:124:21)
    at async ModuleJob.run (node:internal/modules/esm/module_job:190:5) {
  page: '/auth/signup'
}
```

### Steps To Reproduce

Upgrade to the latest refine packages and run next.js app.

### Expected behavior

No error

### Packages

- @refinedev/devtools "1.1.37"

### Additional Context

_No response_

Hey, I had to manually set the dependency version to `0.4.1` in the package-lock.json file for this to work, deleting the file was not enough for the npm install command to fix this.

I am getting another error still though:
```sh
⨯ Error: Cannot find module '/Users/Projects/Code/app/node_modules/next/router' imported from /Users/Projects/Code/app/node_modules/@refinedev/nextjs-router/dist/pages.mjs
Did you mean to import next/router.js?
    at new NodeError (node:internal/errors:405:5)
    at finalizeResolution (node:internal/modules/esm/resolve:327:11)
    at moduleResolve (node:internal/modules/esm/resolve:946:10)
    at defaultResolve (node:internal/modules/esm/resolve:1132:11)
    at nextResolve (node:internal/modules/esm/loader:163:28)
    at ESMLoader.resolve (node:internal/modules/esm/loader:835:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:424:18)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:77:40)
    at link (node:internal/modules/esm/module_job:76:36) {
  code: 'ERR_MODULE_NOT_FOUND',
  page: '/_next/static/webpack/bd114e6d6f2223a0.webpack.hot-update.json'
}
``

--- a/.changeset/rude-suns-agree.md
+++ b/.changeset/rude-suns-agree.md
@@ -1,0 +1,10 @@
+---
+"@refinedev/nextjs-router": patch
+"@refinedev/react-router-v6": patch
+"@refinedev/remix-router": patch
+"@refinedev/strapi-v4": patch
+---
+
+fix: replace imports of `qs` with default imports
+
+Updated `qs` imports and usage to prevent issues with ESM builds and to ensure correctly importing the module.

--- a/packages/nextjs-router/src/app/bindings.tsx
+++ b/packages/nextjs-router/src/app/bindings.tsx
@@ -7,7 +7,7 @@ import {
 } from "@refinedev/core";
 import { useRouter, usePathname, useSearchParams } from "next/navigation";
 import Link from "next/link";
-import { parse, stringify } from "qs";
+import qs from "qs";
 import React, { ComponentProps, useContext } from "react";
 import { paramsFromCurrentPath } from "../common/params-from-current-path";
 import { convertToNumberIfPossible } from "src/common/convert-to-number-if-possible";
@@ -46,7 +46,7 @@ export const routerBindings: RouterBindings = {
 
         const urlQuery = {
           ...(keepQuery
-            ? parse(searchParamsObj.toString(), {
+            ? qs.parse(searchParamsObj.toString(), {
                 ignoreQueryPrefix: true,
               })
             : {}),
@@ -65,7 +65,7 @@ export const routerBindings: RouterBindings = {
         const hasUrlQuery = Object.keys(urlQuery).length > 0;
 
         const fullPath = `${urlTo}${
-          hasUrlQuery ? stringify(urlQuery, stringifyConfig) : ""
+          hasUrlQuery ? qs.stringify(urlQuery, stringifyConfig) : ""
         }${hasUrlHash ? urlHash : ""}`;
 
         if (type === "path") {
@@ -109,7 +109,7 @@ export const routerBindings: RouterBindings = {
 
     const parsedParams = React.useMemo(() => {
       const searchParams = searchParamsObj.toString();
-      return parse(searchParams, { ignoreQueryPrefix: true });
+      return qs.parse(searchParams, { ignoreQueryPrefix: true });
     }, [searchParamsObj]);
 
     const fn = React.useCallback(() => {

--- a/packages/nextjs-router/src/common/parse-table-params.ts
+++ b/packages/nextjs-router/src/common/parse-table-params.ts
@@ -1,8 +1,8 @@
-import { parse } from "qs";
+import qs from "qs";
 import type { ParsedParams } from "@refinedev/core";
 
 const parseTableParams = (search: string) => {
-  const parsed: ParsedParams = parse(search, { ignoreQueryPrefix: true });
+  const parsed: ParsedParams = qs.parse(search, { ignoreQueryPrefix: true });
 
   const tableReady = {
     ...parsed,

--- a/packages/nextjs-router/src/pages/bindings.tsx
+++ b/packages/nextjs-router/src/pages/bindings.tsx
@@ -7,7 +7,7 @@ import {
 } from "@refinedev/core";
 import { useRouter } from "next/router";
 import Link from "next/link";
-import { parse, stringify } from "qs";
+import qs from "qs";
 import React, { ComponentProps, useContext } from "react";
 import { paramsFromCurrentPath } from "../common/params-from-current-path";
 import { convertToNumberIfPossible } from "src/common/convert-to-number-if-possible";
@@ -44,7 +44,7 @@ export const routerBindings: RouterBindings = {
 
         const urlQuery = {
           ...(keepQuery
-            ? parse(pathname.split("?")[1], {
+            ? qs.parse(pathname.split("?")[1], {
                 ignoreQueryPrefix: true,
               })
             : {}),
@@ -63,7 +63,7 @@ export const routerBindings: RouterBindings = {
         const hasUrlQuery = Object.keys(urlQuery).length > 0;
 
         const fullPath = `${urlTo}${
-          hasUrlQuery ? stringify(urlQuery, stringifyConfig) : ""
+          hasUrlQuery ? qs.stringify(urlQuery, stringifyConfig) : ""
         }${hasUrlHash ? urlHash : ""}`;
 
         if (type === "path") {
@@ -109,11 +109,11 @@ export const routerBindings: RouterBindings = {
 
     const parsedParams = React.useMemo(() => {
       const searchParams = pathname.split("?")[1];
-      return parse(searchParams, { ignoreQueryPrefix: true });
+      return qs.parse(searchParams, { ignoreQueryPrefix: true });
     }, [pathname]);
 
     const fn = React.useCallback(() => {
-      const parsedQuery = parse(query as Record<string, string>, {
+      const parsedQuery = qs.parse(query as Record<string, string>, {
         ignoreQueryPrefix: true,
       });
       const combinedParams = {

--- a/packages/react-router-v6/src/bindings.tsx
+++ b/packages/react-router-v6/src/bindings.tsx
@@ -7,7 +7,7 @@ import {
   ResourceContext,
 } from "@refinedev/core";
 import { useCallback, useContext } from "react";
-import { parse, stringify } from "qs";
+import qs from "qs";
 import {
   useNavigate,
   useLocation,
@@ -42,7 +42,7 @@ export const routerBindings: RouterBindings = {
         const urlQuery = {
           ...(keepQuery &&
             existingSearch &&
-            parse(existingSearch, { ignoreQueryPrefix: true })),
+            qs.parse(existingSearch, { ignoreQueryPrefix: true })),
           ...query,
         };
 
@@ -63,7 +63,7 @@ export const routerBindings: RouterBindings = {
         const urlTo = to || "";
 
         const fullPath = `${urlTo}${
-          hasUrlQuery ? stringify(urlQuery, stringifyConfig) : ""
+          hasUrlQuery ? qs.stringify(urlQuery, stringifyConfig) : ""
         }${hasUrlHash ? urlHash : ""}`;
 
         if (type === "path") {
@@ -104,7 +104,7 @@ export const routerBindings: RouterBindings = {
     }
 
     const fn = useCallback(() => {
-      const parsedSearch = parse(search, { ignoreQueryPrefix: true });
+      const parsedSearch = qs.parse(search, { ignoreQueryPrefix: true });
 
       const combinedParams = {
         ...params,

--- a/packages/remix/src/bindings.tsx
+++ b/packages/remix/src/bindings.tsx
@@ -6,7 +6,7 @@ import {
   ParseResponse,
 } from "@refinedev/core";
 import { useParams, useLocation, useNavigate, Link } from "@remix-run/react";
-import { parse, stringify } from "qs";
+import qs from "qs";
 import React, { ComponentProps, useCallback, useContext } from "react";
 import { paramsFromCurrentPath } from "./params-from-current-path";
 import { convertToNumberIfPossible } from "./convert-to-number-if-possible";
@@ -36,7 +36,7 @@ export const routerBindings: RouterBindings = {
         const urlQuery = {
           ...(keepQuery &&
             existingSearch &&
-            parse(existingSearch, { ignoreQueryPrefix: true })),
+            qs.parse(existingSearch, { ignoreQueryPrefix: true })),
           ...query,
         };
 
@@ -57,7 +57,7 @@ export const routerBindings: RouterBindings = {
         const urlTo = to || "";
 
         const fullPath = `${urlTo}${
-          hasUrlQuery ? stringify(urlQuery, stringifyConfig) : ""
+          hasUrlQuery ? qs.stringify(urlQuery, stringifyConfig) : ""
         }${hasUrlHash ? urlHash : ""}`;
 
         if (type === "path") {
@@ -98,7 +98,7 @@ export const routerBindings: RouterBindings = {
     const inferredId = inferredParams.id;
 
     const fn = useCallback(() => {
-      const parsedSearch = parse(search, { ignoreQueryPrefix: true });
+      const parsedSearch = qs.parse(search, { ignoreQueryPrefix: true });
 
       const combinedParams = {
         ...inferredParams,

--- a/packages/remix/src/parse-table-params.ts
+++ b/packages/remix/src/parse-table-params.ts
@@ -1,8 +1,8 @@
-import { parse } from "qs";
+import qs from "qs";
 import type { ParsedParams } from "@refinedev/core";
 
 export const parseTableParams = (search: string) => {
-  const parsed: ParsedParams = parse(search, { ignoreQueryPrefix: true });
+  const parsed: ParsedParams = qs.parse(search, { ignoreQueryPrefix: true });
 
   const tableReady = {
     ...parsed,

--- a/packages/strapi-v4/src/dataProvider.ts
+++ b/packages/strapi-v4/src/dataProvider.ts
@@ -1,6 +1,6 @@
 import { DataProvider as IDataProvider, HttpError } from "@refinedev/core";
 import { AxiosInstance } from "axios";
-import { stringify } from "qs";
+import qs from "qs";
 import {
   axiosInstance,
   generateFilter,
@@ -41,7 +41,7 @@ export const DataProvider = (
     };
 
     const { data } = await httpClient.get(
-      `${url}?${stringify(query, {
+      `${url}?${qs.stringify(query, {
         encodeValuesOnly: true,
       })}&${queryFilters}`,
     );
@@ -78,7 +78,7 @@ export const DataProvider = (
     };
 
     const { data } = await httpClient.get(
-      `${url}?${stringify(query, {
+      `${url}?${qs.stringify(query, {
         encodeValuesOnly: true,
       })}&${queryFilters}`,
     );
@@ -199,7 +199,7 @@ export const DataProvider = (
       publicationState,
     };
 
-    const url = `${apiUrl}/${resource}/${id}?${stringify(query, {
+    const url = `${apiUrl}/${resource}/${id}?${qs.stringify(query, {
       encode: false,
     })}`;
 
@@ -248,7 +248,7 @@ export const DataProvider = (
     if (sorters) {
       const sortQuery = generateSort(sorters);
       if (sortQuery.length > 0) {
-        requestUrl = `${requestUrl}&${stringify({
+        requestUrl = `${requestUrl}&${qs.stringify({
           sort: sortQuery.join(","),
         })}`;
       }
@@ -260,7 +260,7 @@ export const DataProvider = (
     }
 
     if (query) {
-      requestUrl = `${requestUrl}&${stringify(query)}`;
+      requestUrl = `${requestUrl}&${qs.stringify(query)}`;
     }
 
     let axiosResponse;

--- a/packages/strapi-v4/src/helpers/auth.ts
+++ b/packages/strapi-v4/src/helpers/auth.ts
@@ -1,6 +1,6 @@
 import { MetaQuery, pickNotDeprecated } from "@refinedev/core";
 import axios from "axios";
-import { stringify } from "qs";
+import qs from "qs";
 
 interface ILoginResponse {
   jwt: string;
@@ -57,7 +57,7 @@ export const AuthHelper = (apiUrl: string) => ({
     };
 
     return await axios.get<IUser>(
-      `${apiUrl}/users/me?${stringify(query, {
+      `${apiUrl}/users/me?${qs.stringify(query, {
         encodeValuesOnly: true,
       })}`,
       {

--- a/packages/strapi-v4/src/utils/generateFilter.ts
+++ b/packages/strapi-v4/src/utils/generateFilter.ts
@@ -1,6 +1,6 @@
 import { CrudFilters, LogicalFilter, ConditionalFilter } from "@refinedev/core";
 import { mapOperator } from "./mapOperator";
-import { stringify, parse } from "qs";
+import qs from "qs";
 
 export const generateNestedFilterField = (field: string) => {
   const fields = field.split(".");
@@ -77,9 +77,9 @@ export const generateFilter = (filters?: CrudFilters) => {
     });
   }
 
-  const parsedQuery = parse(rawQuery, { depth: 15 });
+  const parsedQuery = qs.parse(rawQuery, { depth: 15 });
 
-  const queryFilters = stringify(parsedQuery, { encodeValuesOnly: true });
+  const queryFilters = qs.stringify(parsedQuery, { encodeValuesOnly: true });
 
   return queryFilters;
 };

